### ISSUE TITLE
Main program does not exit cleanly

### ISSUE THREAD

**Issue by subakun** [theolind/pymysensors#153]
[branch master - version 0.14.0 - Using serial gateway - NO asyncio]
After calling the SerialGateway.stop() method the program does not return to console but seems to be looping in a still alive thread ( probably the persistence thread).

************************************************************************************
```py
MYSGW_Serial_Port = '/dev/ttyMSGW'
....
GATEWAY = mysensors.SerialGateway(
MYSGW_Serial_Port, event_callback=event, persistence=True,
persistence_file='./mysensors.json', protocol_version='2.0', baud=115200,
timeout=1.0, reconnect_timeout=10.0)
GATEWAY.start_persistence()
GATEWAY.start()
....
....
GATEWAY.stop() #-> main thread does not go past this point
exit(0)
```
*************************************************************************************

**Comment by MartinHjelmare** [comment]
I think I know what the problem is. The function to cancel the persistence thread gets overwritten after the second save. We should move the persistence scheduler to the Threading gateway base class, and store the cancel function in an instance attribute on each save, like is done in the async gateway base class.

I'll try to make it happen during the weekend. But PR is also welcome.

**Comment by subakun** [comment]
Martin many thanks for your reply.
what is PR ?

**Comment by MartinHjelmare** [comment]
PR is short for pull request in github and is a proposal for code change, recorded in a git branch.

**Comment by MartinHjelmare** [comment]
Fixed in #154.
